tests/integration/buildcmd/test_build_cmd.py

```diff
@@ -1812,3 +1812,54 @@ def test_nested_build(self, use_container, cached, parallel):
                     ("LocalNestedStack/Function2", {"pi": "3.14"}),
                 ],
             )
+
+
+@skipIf(
+    ((IS_WINDOWS and RUNNING_ON_CI) and not CI_OVERRIDE),
+    "Skip build tests on windows when running in CI unless overridden",
+)
+class TestBuildWithS3FunctionsOrLayers(NestedBuildIntegBase):
+    # Test Suite where `BuildMethod` is explicitly specified.
+
+    template = "template-with-s3-code.yaml"
+    EXPECTED_FILES_PROJECT_MANIFEST = {
+        "__init__.py",
+        "main.py",
+        "numpy",
+        # 'cryptography',
+        "requirements.txt",
+    }
+
+    # @pytest.mark.flaky(reruns=3)
+    def test_functions_layers_with_s3_codeuri(self):
+        if SKIP_DOCKER_TESTS:
+            self.skipTest(SKIP_DOCKER_MESSAGE)
+
+        """
+        Build template above and verify that each function call returns as expected
+        """
+        cmdlist = self.get_command_list(
+            use_container=True,
+        )
+
+        LOG.info("Running Command: %s", cmdlist)
+        LOG.info(self.working_dir)
+
+        command_result = run_command(cmdlist, cwd=self.working_dir)
+
+        if not SKIP_DOCKER_TESTS:
+            self._verify_build(
+                ["ServerlessFunction", "LambdaFunction"],
+                [""],  # there is only one stack
+                command_result,
+            )
+
+            overrides = self._make_parameter_override_arg({})
+            self._verify_invoke_built_functions(
```

Comment:
The template is buildable because s3 based locations are explicitly skipped, what does the local invoke do here?

Reply:
good point. I don't think functions containing these remote layers should be local invocable at all. What do you think, disallow their invocation? @sriram-mv

Reply:
I implemented this integ test base in #2760, will rebase this PR once #2760 is merged.

Reply:
Correct: these two functions are still invocable

Reply:
maybe only test building? I feel that testing invoke might be unnecessary here

Reply:
no strong opinion here, the invoke is added based on the current sam-cli behavior, we might not need this in the future.  here removed

Reply:
does the invoke just skip? what is our current behavior? could we add a comment on what the behavior we are trying to test. This will be useful for future deep dives.